components/scream/cmake/ScreamUtils.cmake

```diff
@@ -30,75 +30,86 @@ macro (SetVarDependingOnTestProfile var_name val_short val_medium val_long)
   endif()
 endmacro()
 
-function(CreateUnitTest test_name test_srcs scream_libs)
-  set(options EXCLUDE_MAIN_CPP SERIAL PRINT_OMP_AFFINITY)
-  set(oneValueArgs TEST_EXE EXE_ARGS DEP)
-  set(multiValueArgs
-    MPI_RANKS THREADS
-    INCLUDE_DIRS
-    COMPILER_DEFS
-    COMPILER_C_DEFS COMPILER_CXX_DEFS COMPILER_F_DEFS
-    COMPILER_FLAGS
-    COMPILER_C_FLAGS COMPILER_CXX_FLAGS COMPILER_F_FLAGS
-    LABELS PROPERTIES)
-  cmake_parse_arguments(PARSE_ARGV 3 CreateUnitTest "${options}" "${oneValueArgs}" "${multiValueArgs}")
-  CheckMacroArgs(CreateUnitTest CreateUnitTest "${options}" "${oneValueArgs}" "${multiValueArgs}")
+set(SCREAM_CUT_EXEC_OPTIONS ${CUT_EXEC_OPTIONS})
+set(SCREAM_CUT_EXEC_1V_ARGS ${CUT_EXEC_1V_ARGS})
+set(SCREAM_CUT_EXEC_MV_ARGS ${CUT_EXEC_MV_ARGS})
+
+set(SCREAM_CUT_TEST_OPTIONS ${CUT_TEST_OPTIONS})
+set(SCREAM_CUT_TEST_1V_ARGS ${CUT_TEST_1V_ARGS})
+set(SCREAM_CUT_TEST_MV_ARGS ${CUT_TEST_MV_ARGS})
+
+#
+# Adjust Ekat CUT options based on SCREAM specifics
+#
+
+# Scream always excludes the ekat test session since it has its own
+list(REMOVE_ITEM SCREAM_CUT_EXEC_OPTIONS EXCLUDE_TEST_SESSION)
+
+# Libs are a position arg for SCREAM, not an optional arg like in EKAT
+list(REMOVE_ITEM SCREAM_CUT_EXEC_MV_ARGS LIBS)
+
+# MPI stuff is set in scream's cache config and is not configurable per-test
+list(REMOVE_ITEM SCREAM_CUT_TEST_1V_ARGS MPI_EXEC_NAME MPI_NP_FLAG)
+list(REMOVE_ITEM SCREAM_CUT_TEST_MV_ARGS MPI_EXTRA_ARGS)
+
+###############################################################################
+function(CreateUnitTestExec exec_name test_srcs scream_libs)
+###############################################################################
+  cmake_parse_arguments(cute "${SCREAM_CUT_EXEC_OPTIONS}" "${SCREAM_CUT_EXEC_1V_ARGS}" "${SCREAM_CUT_EXEC_MV_ARGS}" ${ARGN})
+  CheckMacroArgs(CreateUnitTestExec cute "${SCREAM_CUT_EXEC_OPTIONS}" "${SCREAM_CUT_EXEC_1V_ARGS}" "${SCREAM_CUT_EXEC_MV_ARGS}")
+
+  separate_cut_arguments(cute "${SCREAM_CUT_EXEC_OPTIONS}" "${SCREAM_CUT_EXEC_1V_ARGS}" "${SCREAM_CUT_EXEC_MV_ARGS}" options)
```

Comment:
Is this really necessary? Can't we just pass `${ARGN}` to `EkatCreateUnitTest`, explicitly adding `EXCLUDE_TEST_SESSION` and `COMPILER_F_FLAGS ${SCREAM_Fortran_FLAGS}`?